docs/reconcilation.md

```diff
@@ -0,0 +1,72 @@
+During a reconciliation "step" the sub-reconciler can decide if the reconciliation can continue or not.
+That means that a sub-reconciler can block the reconciliation from completion.
+
+### How to debug a stuck reconciliation
+
+A reconciliation can get stuck if the desired state can't be reached e.g. because an upstream resource won't be available (Pod is stuck in `Pending`).
+There are two places in the `FoundationDBCluster` status that can provide further guidance on why the reconciliation is stuck:
+
+- The [generations](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.33.0/api/v1beta1/foundationdbcluster_types.go#L670) that describe the latest reconciled generation (if present) and any additional generations (and the reason) that are not fully reconciled. Most of the conditions are fairly well describing.
+- The [process group conditions](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.33.0/api/v1beta1/foundationdbcluster_types.go#L622-L642) that describes for each process group the current conditions. If a process group has no conditions that means that the process group is fully reconciled. In most cases a process group reflects a single instance but for the case with multiple Storage Servers per disk we would have multiple instances for a process group. The conditions should give you an idea which process group are blocking the reconciliation and why e.g. `IncorrectCommandLine` would indicate that the process is running with a different commandline than expected by the controller.
+
+The following steps should help to get the reconciliation unstuck:
+
+1. Check the `generations` to get an overview why the reconciliation is stuck.
+1. Check the `process group conditions` to see which proces actually block the reconcilation.
+1. Check the upstream resouces e.g. Pods (are they running).
+
+### Examples
+
+Assume you have a cluster `sample-cluster` with a stuck reconciliation.
+You check the `generations` and find out that the latest generation has `hasUnhealthyProcess`.
+In the next step you check the `process group conditions` and find out that the instance `sample-cluster-storage-1` has the condition `MissingProcesses`.
+Now you can check the Pod for the instance with `kubectl get po -l fdb-instance-id=sample-cluster-storage-1,fdb-cluster-name=sample-cluster` (or you use `kubectl describe`).
+If the Pod is in a `Running` state check the Node of the Pod for potential network issues and replace the instance.
+If the Pod is in a `Terminating`, `Failed` or `Failed` state check the status of the Pod e.g. with `kubectl descibe` or `kubectl get po ... -oyaml` for the reason, depending on the reason contact the Kube team or replace the instance.
+
+### Automatic replacements
+
+The FDB operator supports [automatic replacements](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/v0.33.0/docs/cluster_spec.md#automaticreplacementoptions) which allows the operator to replace failed Pods automatically.
```

Comment:
I would like to keep it a specific version to pin point to the specific place otherwise once we change the file and the line numbers will change we point to an unrelated place. With the fixed version at least the reader should be able to find the according place in the latest (or specific) version of the operator.